packages/rich-text/src/component/editable.js

```diff
@@ -160,20 +154,42 @@ export default class Editable extends Component {
 	render() {
 		const {
 			tagName = 'div',
-			style,
+			style = {},
 			record,
 			valueToEditableHTML,
-			className: additionalClassName,
+			className,
 			isPlaceholderVisible,
 			...remainingProps
 		} = this.props;
 
 		delete remainingProps.setRef;
 
+		// In HTML, leading and trailing spaces are not visible, and multiple
+		// spaces elsewhere are visually reduced to one space. This rule
+		// prevents spaces from collapsing so all space is visible in the editor
+		// and can be removed.
+		// It also prevents some browsers from inserting non-breaking spaces at
+		// the end of a line to prevent the space from visually disappearing.
+		// Sometimes these non breaking spaces can linger in the editor causing
+		// unwanted non breaking spaces in between words. If also prevent
+		// Firefox from inserting a trailing `br` node to visualise any trailing
+		// space, causing the element to be saved.
+		//
+		// > Authors are encouraged to set the 'white-space' property on editing
+		// > hosts and on markup that was originally created through these
+		// > editing mechanisms to the value 'pre-wrap'. Default HTML whitespace
+		// > handling is not well suited to WYSIWYG editing, and line wrapping
+		// > will not work correctly in some corner cases if 'white-space' is
+		// > left at its default value.
+		// >
+		// > https://html.spec.whatwg.org/multipage/interaction.html#best-practices-for-in-page-editors
+
+		style.whiteSpace = 'pre-wrap';
```

Comment:
Is it ok to mutate the prop here. I assume we're the only consumer component of the component, so it should be fine but wondering if we should avoid it if ever this becomes a more public component.

Reply:
Is this still necessary with this https://github.com/WordPress/gutenberg/pull/15212/files#diff-a4fd2f97aed7d4f0c5f8480292ef5f92R115 

Why the redundancy?

Reply:
Still necessary right? `shouldComponentUpdate` is not called on first render.

Right, I shouldn't mutate. :D